tests/dummy/snippets/ts/yield/ember-concurrency-async.d.ts

```diff
@@ -0,0 +1,25 @@
+import { Task, task, taskGroup } from 'ember-concurrency';
+import { taskFor } from './utils';
+
+type AsyncTaskFunction<T, Args extends any[]> = (...args: Args) => Promise<T>;
+
+type AnyAsyncTaskFunction = AsyncTaskFunction<any, any[]>;
+
+type Args<T extends AnyAsyncTaskFunction> =
+  T extends (...args: infer A) => Promise<any> ? A : [];
+
+type Return<T extends AnyAsyncTaskFunction> =
+  T extends (...args: any[]) => Promise<infer R> ? R : unknown;
+
+declare module 'ember-concurrency' {
+  function task<T extends AnyAsyncTaskFunction>(taskFn: T):
+    TaskProperty<Return<T>, Args<T>>;
+
+  function taskGroup<T extends AnyAsyncTaskFunction>(taskFn: T):
+    TaskGroup<Return<T>>;
+}
```

Comment:
Is there a good way for ember-concurrency-async to bundle this monkey-patch? I could include a `.d.ts` file, but I'm not sure if TypeScript will ever "finds" it, since most people have `node_modules` on "excludes" and you don't otherwise need to import anything else from the addon to use it.

Reply:
Still relevant cc @chriskrycho @dfreeman

Reply:
The typical pattern is to include it in `index.d.ts` or whatever your `types` entrypoint is (or a module imported from there), and then instruct users to write e.g. `import 'ember-concurrency-async';` in a central place in their project.

Reply:
I can do that (and update this documentation) once this has been released

Reply:
https://github.com/chancancode/ember-concurrency-async/pull/2